commons-test/src/main/java/org/infinispan/commons/test/TestSuiteProgress.java

```diff
@@ -13,66 +13,79 @@
+   private final PrintStream out;
+   private final boolean useColor;
 
-   static void testStarted(String name) {
+   public TestSuiteProgress() {
+      // Use a system property to avoid color
+      useColor = !Boolean.getBoolean("ansi.strip");
+      out = System.out;
+   }
+
+   void testStarted(String name) {
       String message = "Test starting: " + name;
-      consoleLog(message);
+      progress(message);
       log.info(message);
    }
 
-   static void testFinished(String name) {
+   void testFinished(String name) {
+      succeeded.incrementAndGet();
       String message = "Test succeeded: " + name;
-      consoleLog(message);
+      progress(message, GREEN);
       log.info(message);
-      succeeded.incrementAndGet();
-      printStatus();
+
    }
 
-   static void testFailed(String name, Throwable exception) {
+   void testFailed(String name, Throwable exception) {
+      failed.incrementAndGet();
       String message = "Test failed: " + name;
-      consoleLog(message);
+      progress(message, RED);
       log.error(message, exception);
-      failed.incrementAndGet();
-      printStatus();
    }
 
-   static void testIgnored(String name) {
+   void testIgnored(String name) {
+      skipped.incrementAndGet();
       String message = "Test ignored: " + name;
-      consoleLog(message);
+      progress(message, YELLOW);
       log.info(message);
-      skipped.incrementAndGet();
-      printStatus();
    }
 
-   static void testAssumptionFailed(String name, Throwable exception) {
+   void testAssumptionFailed(String name, Throwable exception) {
+      skipped.incrementAndGet();
       String message = "Test assumption failed: " + name;
-      consoleLog(message);
+      progress(message, YELLOW);
       log.info(message, exception);
-      skipped.incrementAndGet();
-      printStatus();
    }
 
-   static void setupFailed(String name, Throwable exception) {
+   void setupFailed(String name, Throwable exception) {
+      failed.incrementAndGet();
       String message = "Test setup failed: " + name;
-      consoleLog(message);
+      progress(message, RED);
       log.error(message, exception);
-      failed.incrementAndGet();
-      printStatus();
    }
 
-   private static void printStatus() {
-      String message = "Tests succeeded: " + succeeded.get() + ", failed: " + failed.get() + ", skipped: " +
-            skipped.get();
-      consoleLog(message);
+   synchronized void progress(CharSequence message) {
+      out.printf("[OK: %5s, KO: %5s, SKIP: %5s] %s\n", succeeded.get(), failed.get(), skipped.get(), message);
+   }
+
+   void progress(String message, String color) {
+      if (useColor) {
+         progress(color + message + RESET);
+      } else
+         progress(message);
    }
 
-   private static void consoleLog(String message) {
-      System.out.println("[" + TestSuiteProgress.class.getSimpleName() + "] " + message);
+   public static String string(char... chars) {
```

Comment:
I guess God did not want us to have a ```String``` constructor taking a variable number of chars for a reason... :)